File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        l=0
        ans = 0
        while l<len(chars):
            count=0
            letter = chars[l]
            while l<len(chars) and chars[l]==letter:
                l+=1
                count+=1
            chars[ans]=letter
            ans+=1
            if count>1:
                for c in str(count):
                    chars[ans]=c
                    ans+=1
            
        return ans
            

File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int left=0;
        int right=0;
        while (right<chars.length){
            char ch = chars[right];
            int count=0;
            while (right<chars.length && chars[right] == ch ) {
                right+=1;
                count+=1;
            }
            chars[left]=ch;
            left+=1;
            if (count>1) {
                for (char c : String.valueOf(count).toCharArray()) {
                    chars[left++] = c;
                }
            }
        }
        return left;
    }
}